pandaharvester/harvestersubmitter/pbs_bf_submitter.py

```diff
@@ -0,0 +1,183 @@
+                    if stdOut is not None:
+                        workSpec.set_log_file('stdout', '{0}/{1}'.format(baseDir, stdOut))
+                    if stdErr is not None:
+                        workSpec.set_log_file('stderr', '{0}/{1}'.format(baseDir, stdErr))
+                tmpRetVal = (True, '')
+            else:
+                # failed
+                errStr = stdOut + ' ' + stdErr
+                tmpLog.error(errStr)
+                tmpRetVal = (False, errStr)
+            retList.append(tmpRetVal)
+        return retList
+
+    # make batch script
+    def make_batch_script(self, workspec):
+        # make logger
+        tmpLog = self.make_logger(baseLogger, 'workerID={0}'.format(workspec.workerID),
+                                  method_name='make_batch_script')
+
+        resource = self.get_max_bf_resources(workspec)
+        if resource:
+            tmpLog.debug("Selected resources: %s" % resource)
+            workspec.nCore = (resource['nodes'] - self.nodesToDecrease) * self.nCorePerNode
+            duration = resource['duration'] - self.durationSecondsToDecrease
+        else:
+            workspec.nCore = self.nCore
+            duration = self.defaultDurationSeconds
+
+        workspec.nCore = self.adjust_needed_resource(workspec)
+        duration = str(datetime.timedelta(seconds=duration))
+        tmpFile = tempfile.NamedTemporaryFile(delete=False, suffix='_submit.sh', dir=workspec.get_access_point())
+        tmpFile.write(self.template.format(nCorePerNode=self.nCorePerNode,
+                                           localQueue=self.localQueue,
+                                           projectName=self.projectName,
+                                           nNode=workspec.nCore / self.nCorePerNode,
+                                           accessPoint=workspec.accessPoint,
+                                           duration=duration,
+                                           workerID=workspec.workerID)
+                      )
+        tmpFile.close()
+        return tmpFile.name
+
+    # get log file names
+    def get_log_file_names(self, batch_script, batch_id):
+        stdOut = None
+        stdErr = None
+        with open(batch_script) as f:
+            for line in f:
+                if not line.startswith('#PBS'):
+                    continue
+                items = line.split()
+                if '-o' in items:
+                    stdOut = items[-1].replace('$SPBS_JOBID', batch_id)
+                elif '-e' in items:
+                    stdErr = items[-1].replace('$PBS_JOBID', batch_id)
+        return stdOut, stdErr
+
+    # get backfill resources
+    def get_bf_resources(self, workspec):
+        # make logger
+        tmpLog = self.make_logger(baseLogger, 'workerID={0}'.format(workspec.workerID),
+                                  method_name='get_bf_resources')
+
+        resources = {}
+        # command
+        comStr = "showbf -p {0}".format(self.partition)
```

Comment:
On Titan "showbf -p titan --blocking"  should be used, otherwise not actual info will be provided